src/utils/utils.ts

```diff
@@ -86,6 +86,11 @@ export const transformEmptyStringToNullInObj = (obj: any) => {
   return obj;
 }
 
+export const removeEmptyStringsDeep = (obj: any) => {
+  Object.keys(obj).forEach(key => (obj[key] && typeof obj[key] === 'object') && removeEmptyStringsDeep(obj[key]) || (obj[key] === '') && delete obj[key]);
```

Comment:
The logic is probably correct but it's kind of hard to follow, could you add some parantheses to better separate the `and`s and `or`s